### PR TITLE
Fix Apache IDN PHP scripts

### LINKED ISSUE #4508: [Bug] "File not found." PHP issues with IDN after update.
### Describe the bug

Issue Summary:

After performing a sudo apt update and sudo apt upgrade, some of my WordPress sites with internationalized domain names (IDNs) containing special characters (like "Ø") started returning a "File not found" error when trying to access /wp-admin or any PHP files. This issue is specific to the IDN domains, while other domains without special characters continue to function normally.

Error Log:

`[Fri Aug 09 12:45:53.431012 2024] [proxy_fcgi:error] [pid 30534:tid 30578] [client IP:0] AH01071: Got error 'Primary script unknown', referer: https://domain.com/wp-admin/`

Troubleshooting:

I've confirmed that the problem is not related to any other domain, or is of another cause. The problem only affects the domains with special characters in their names.

### Tell us how to replicate the bug

1. Add an IDN such as 'ørne.dk'
2. Try accessing a php file on that domain

### Which components are affected by this bug?

(Backend) Web Server (Nginx, Apache2)

### Hestia Control Panel Version

1.8.11

### Operating system

Ubuntu 22

### Log capture

```shell
[Fri Aug 09 12:45:53.431012 2024] [proxy_fcgi:error] [pid 30534:tid 30578] [client IP:0] AH01071: Got error 'Primary script unknown', referer: https://domain.com/wp-admin/
```

### PR DESCRIPTION
**Pull request by divinity76**
Fix GH-4508
Fix a problem when using nginx->apache->php-fpm where unicode domains does not execute php files.

this replace
```
    <FilesMatch \.php$>
        SetHandler "proxy:unix:/run/php/php8.2-fpm-idn-tést.eu.sock|fcgi://localhost"
    </FilesMatch>"
```
with

```
ProxyPassMatch ^/(.*\.php)(?:$|\?) "unix:/run/php/php8.2-fpm-idn-tést.eu.sock|fcgi://localhost/home/testhoster1/web/idn-tést.eu/public_html/$1" nocanon
```

This is 1 of 2 proposed fixes.
There is an alternative proposal explained in detail at https://github.com/hestiacp/hestiacp/issues/4508#issuecomment-2380014131 , but this fix seems OK to me.

Also the 2 proposals are not mutually exclusive, we can still implement the second proposal in the future.

Here is what the upgrade script git diffs looks like:
image

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,13 +1,16 @@
 {
 	"editor.defaultFormatter": "esbenp.prettier-vscode",
 	"[php]": {
-		"editor.defaultFormatter": "esbenp.prettier-vscode"
+		"editor.defaultFormatter": "bmewburn.vscode-intelephense-client"
```
Comment (AlecRust) [inline .vscode/settings.json]: Why remove the Prettier plugin?

Comment (divinity76) [reply .vscode/settings.json]: oops I didn't mean to touch this file at all. reverting

**Comment by divinity76** [comment]
I asked for people to help test this at https://github.com/hestiacp/hestiacp/issues/4508#issuecomment-2388007039 and [@ctiger did](https://github.com/hestiacp/hestiacp/issues/4508#issuecomment-2388412217):


>Had to reinstall the server again, installed the panel using Apache and nginx+php-fpm. I created a fake IDN domain `этотдоменнесуществует.рф`, renamed index.html to index.php and applied your patch.
**It worked.**
> _I'll try to install WorPress next and check the internal pages.
> Upd: checked, worked

sounds good! 👍

File: install/upgrade/versions/1.9.0_extras/migrate_SetHandler_to_ProxySetMatch.php [hunk 2]
```diff
@@ -0,0 +1,159 @@
+<?php
```
Comment (jaapmarcus) [inline install/upgrade/versions/1.9.0_extras/migrate_SetHandler_to_ProxySetMatch.php]: I think we allready rebuild all websites allready for 1.9 so those files will get replaced in the next step after this step is completed

https://github.com/hestiacp/hestiacp/blob/8d5bd57c8fc5756a8664d15f9a69478df413b45c/install/upgrade/versions/1.9.0.sh#L23

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

File: install/upgrade/versions/1.9.0_extras/migrate_SetHandler_to_ProxySetMatch.php (hunk 2 above)
Comment (divinity76) [reply install/upgrade/versions/1.9.0_extras/migrate_SetHandler_to_ProxySetMatch.php]: oh dang. that was a lot of wasted effort 😅

ok removed.

**Review by ScIT-Raphael — APPROVED** [review]

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**